File: CocoatainerGuiExample/CocoatainerGuiExample/GuiCocoaMug/Kettle.m
//
//  Kettle.m
//  CocoatainerExample
//
//  Created by Jeffrey Bakker on 2015-05-13.
//  Copyright (c) 2015 Jeffrey Bakker. All rights reserved.
//

#import "Kettle.h"

#define FULLCUP 250

@interface Kettle ()
{
@private
    id<ILog> _log;
}
@end

@implementation Kettle

-(id)initWithLog:(id<ILog>)log
{
    self = [super init];
    if (self)
    {
        _log = log;
    }
    return self;
}

-(void)dealloc
{
    [_log write:@"This water got cold and looks old. I will dump it out."];
}

-(void)start
{
    [self heat];
}

-(void)heat
{
    [_log write:@"Boiling water to 100 degrees C."];
}

-(NSInteger)pourCup
{
    [_log write:@"Pouring a cup of hot water."];
    return FULLCUP;
}

@end

File: CocoatainerSwiftExample/CocoatainerSwiftExample/SwiftCocoaMug/Kettle.swift
//
//  Kettle.swift
//  CocoatainerSwiftExample
//
//  Created by Jeffrey Bakker on 2015-05-20.
//  Copyright (c) 2015 Jeffrey Bakker. All rights reserved.
//

import Cocoatainer

public class Kettle: NSObject, HotWaterSource, CCTStartable
{
    let FullCup = 250

    deinit
    {
        print("This water got cold and looks old. I will dump it out.")
    }

    public func start()
    {
        heat()
    }

    public func heat()
    {
        print("Boiling water to 100 degrees C.")
    }

    public func pourCup() -> Int
    {
        print("Pouring a cup of hot water.")
        return FullCup
    }
}
